File: h/合并两个排序的链表.java
/*
public class ListNode {
    int val;
    ListNode next = null;
 
    ListNode(int val) {
        this.val = val;
    }
}*/
public class Solution {
	
    public ListNode Merge(ListNode list1, ListNode list2) {
        if (list1 == null) {
            return list2;
        }
        
        if (list2 == null) {
            return list1;
        }
        
        ListNode pList = null;
		
        if (list1.val < list2.val) {
            pList = list1;
            pList.next = Merge(list1.next, list2);
            
        } else {
            pList = list2;
            pList.next = Merge(list1, list2.next);
        }
		
        return pList;
	}
	
}

File: z/合并两个排序的链表.cc
/*
struct ListNode {
    int val;
    struct ListNode *next;
    ListNode(int x) :
            val(x), next(NULL) {
    }
};*/
class Solution {
 public:
  ListNode* Merge(ListNode* pHead1, ListNode* pHead2) {
    ListNode* dummy = new ListNode(0);
    ListNode* head = dummy;
    while (pHead1 != NULL && pHead2 != NULL) {
      if (pHead1->val < pHead2->val) {
        head->next = pHead1;
        head = head->next;
        pHead1 = pHead1->next;
      } else {
        head->next = pHead2;
        head = head->next;
        pHead2 = pHead2->next;
      }
    }
    head->next = pHead1 != NULL ? pHead1 : pHead2;
    head = dummy->next;
    delete dummy;
    return head;
  }
};
